src/main/java/org/checkerframework/checker/builder/qual/CalledMethodsPredicate.java

```diff
@@ -0,0 +1,34 @@
+package org.checkerframework.checker.builder.qual;
+
+import org.checkerframework.framework.qual.SubtypeOf;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * This annotation represents a predicate on @CalledMethods
+ * annotations that must be true. It is intended to extend
+ * the possible LUB of the @CalledMethods type system to
+ * permit methods to be annotated to require a disjunction
+ * of method calls. So, for instance, if it was acceptable
+ * to call method a() or method b() before calling method c(),
+ * then c()'s receiver should have an @CalledMethodsPredicate
+ * annotation with the argument "a || b"
+ *
+ * The argument is a string. The string must be constructed
+ * from the following grammar:
+ *
+ * S → method name | (S) | S && S | S || S
```

Comment:
If you have `||` and `&&` but not negation, it essentially amounts to enumerating all legal combinations of builder methods. Are you planning to have negation as well? Otherwise, it would be more intuitive to just have lists of lists.

Reply:
I think, to not overcomplicate things, you would be better of if you just allow folks to enumerate legal combinations. Otherwise, we can rathole on things that look cool but have no real value for the paper (e.g., enabling things like `if you use X you must not use Y`)

Reply:
The spec for the EC2 API is going to get really nasty if we have to enumerate all valid combinations, so I don't think I agree with you on this.

I suspect there will be even more complex requirements than that that we will want to specify, and the predicates make that somewhat more reasonable.

I have no intention to support negation. The predicates are just syntactic sugar for a list of lists.

Reply:
Given the spec written in `DescribeImages.astub` I think it's worthwhile to support `&&` and `||` as opposed to just a list of lists

Reply:
Ah, I see what you are saying. So it's more like a regular expression than actual logic. Fine by me, I guess.